### PR TITLE
[FEATURE] Implemented variant attributes api

### PR DESCRIPTION
**Pull request by turbo-ele**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | https://github.com/sulu/SuluProductBundle/issues/315
| License | MIT

#### What's in this PR?

get fields, cget action, post action and delete action for variant attributes relation

### CODE REVIEW

File: Controller/ProductVariantAttributeController.php [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+use Sulu\Component\Security\SecuredControllerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller makes adding and removing of product variants accessible through a REST API.
+ */
+class ProductVariantAttributeController extends RestController implements ClassResourceInterface, SecuredControllerInterface
+{
+    protected static $productEntityName = 'SuluProductBundle:Product';
```
Comment (wachterjohannes) [inline Controller/ProductVariantAttributeController.php]: you could use `Product::class` instead.

File: Product/ProductVariantAttributeManager.php [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\DoctrineJoinDescriptor;
+
+/**
+ * This manager is responsible for handling product attributes.
+ */
+class ProductVariantAttributeManager
+{
+    use ArrayDataTrait;
+
+    private static $attributeEntityName = 'SuluProductBundle:Attribute';
```
Comment (wachterjohannes) [inline Product/ProductVariantAttributeManager.php]: `Attribute::class`

File: Product/ProductVariantAttributeManager.php [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    /**
+     * @var AttributeRepositoryInterface
+     */
+    private $attributeRepository;
+
+    /**
+     * @param ProductRepositoryInterface $productRepository
+     * @param AttributeRepositoryInterface $attributeRepository
+     */
+    public function __construct(
```
Comment (wachterjohannes) [inline Product/ProductVariantAttributeManager.php]: does this fit into a single line?

File: Product/ProductVariantAttributeManager.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+     *
+     * @param string $locale
+     *
+     * @return DoctrineFieldDescriptor[]
+     */
+    public function retrieveFieldDescriptors($locale)
+    {
+        $fieldDescriptors = [];
+
+        $fieldDescriptors['id'] = new DoctrineFieldDescriptor(
```
Comment (wachterjohannes) [inline Product/ProductVariantAttributeManager.php]: why not using xml definition?

File: Controller/ProductVariantAttributeController.php (hunk 1 above)
Comment (turbo-ele) [inline Controller/ProductVariantAttributeController.php]: Unfortunately the list builder depends on this naming. In the (hopefully) near future it's planned to use Sulu-Persistence and replace all the entity names here, with the correct (inheritable) entity namings

File: Product/ProductVariantAttributeManager.php (hunk 3 above)
Comment (turbo-ele) [inline Product/ProductVariantAttributeManager.php]: no

**Pull request merged**